src/fidasim.f90

```diff
@@ -10073,36 +11110,38 @@ subroutine pnpa_mc
                 call hit_npa_detector(ri, vi ,det, rf)
                 if(det.eq.0) cycle gamma_loop
 
-                !! Get beam grid indices at ri
-                call get_indices(ri,ind)
-
                 !! Calculate CX probability with beam and halo neutrals
                 call get_plasma(plasma, pos=ri)
                 call bt_cx_rates(plasma, plasma%denn, vi, beam_ion, rates)
                 if(sum(rates).le.0.) cycle gamma_loop
 
                 !! Weight CX rates by ion source density
-                states=rates*fast_ion%weight/ngamma
+                if(particles%axisym) then
+                    states=rates*fast_ion%weight/(fast_ion%delta_phi/(2*pi)) &
+                           *(pass_grid%nphi*pass_grid%dphi/(2*pi))  &
+                           *beam_grid%dv/(fast_ion%r*pass_grid%dv)/ngamma
+                else
+                    states=rates*fast_ion%weight*beam_grid%dv/(fast_ion%r*pass_grid%dv)/ngamma
+                endif
 
                 !! Attenuate states as the particle moves though plasma
                 call attenuate(ri,rf,vi,states)
 
                 !! Store NPA Flux
-                flux = sum(states)*beam_grid%dv
+                flux = sum(states)*(fast_ion%r*pass_grid%dv)
                 call store_npa(det,ri,rf,vi,flux,fast_ion%class,passive=.True.)
             endif
         enddo gamma_loop
     enddo loop_over_fast_ions
     !$OMP END PARALLEL DO
 
-    npart = pnpa%npart
 #ifdef _MPI
-    call parallel_sum(npart)
+    call parallel_sum(pnpa%npart)
```

Comment:
again you the parallel sum is already done in the `write_npa` subroutine. This will break it.